gh-docs/message-processor/Admin_Services.md

```diff
@@ -0,0 +1,58 @@
+# How to create admin-services for the Message Processor
+
+### Creation Flow
+
+1. Define your Admin Service function
+2. Generate the stub using wsdl
+3. Define function in MessageProcessorAdminServiceClient to call Admin Service function
+4. Call the function from the Message Processor UI jsp
+
+### 1. Defining your Admin Service
+Message Processor Admin Services are defined in the 'MessageProcessorAdminService.java' class in the carbon-mediation repo.
+
+Go to `<repo-home>/components/mediation-admin/org.wso2.carbon.message.processor/` and navigate to the 'service' directory
+to find the 'MessageProcessorAdminService.java' class.
+
+Once you define your function, you will have to generate the MessageProcessorAdminServiceStub file. 
+
+### 2. Generating the MessageProcessorAdminServiceStub using wsdl 
+To generate the stub file, the WSDL file in `<repo-home>/service-stubs/mediation-admin/org.wso2.carbon.message.processor.stub/`
+needs to be edited and then built to generate the stub. 
+
+#### To obtain the contents of the WSDL
+
+ 1. Build 'MessageProcessorAdminService.java' using 'mvn clean install' on the IntelliJ console. 
+ 
+ 2. Copy the generated '.jar' file found in `<repo-home>/components/mediation-admin/org.wso2.carbon.message.processor/target/org.wso2.carbon.message.processor-4.6.106-SNAPSHOT.jar`
+ 
+ 3. Paste the '.jar' file into the 'patches' folder in the product-ui 
```

Comment:
don't we have to create a subdirectory inside patches directory?

Reply:
Yes. I will mention that